lib/tasks/data_migrations.rake

```diff
@@ -0,0 +1,15 @@
+namespace :data_migrations do
+  desc 'Sets all metrics with no useful yes/no responses to have null for satisfaction'
+  task satisfaction_defaults_to_null: :environment do
+    puts "Setting default satisfaction to nil for pages with no responses"
+
+    zero_responses = Facts::Metric.where('useful_yes = 0').where('useful_no = 0')
+
+    zero_responses.in_batches.each_with_index do |metrics, batch_index|
+      puts "Updating #{batch_index} records"
+      metrics.update_all(satisfaction: nil)
+    end
```

Comment:
Going through in batches is going to be really slow given the number of records in this table, I think the default batch size is 1000, so given that, do you know how many batches are required?

This should work without any batching, and thus should be a lot quicker:
``` ruby
Facts::Metric
  .where(useful_yes: 0, useful_no: 0)
  .update_all(satisfaction: nil)
```

Reply:
I am interested in getting your thoughts on how we should tackle these migrations. 

I find value in having a longer rake task with a progress indicator that takes say 30 minutes vs a SQL query that takes 20 silently. 

We could explore also creating a temporary index for queries that expand all the metrics table, then do the change, and then remove the index. 

I am happy with this to go through, but It is not obvious for me that this is a better solution, and it would be nice to explore other options for the future.

cc/ @chao-xian @theseanything @cbaines

Reply:
I had difficulty testing an index locally as I did try adding it and then running the migration, but my VM ran out of disk space...